File: ioi2020-mushrooms/solution/trivial-one-by-one.java
/*
 * Trivial one-by-one solution
 *
 * Author: Kian Mirjalali
 */
public class mushrooms {

	private int use_machine(int[] x) {
		return stub.use_machine(x);
	}

	public int count_mushrooms(int n) {
		int s = 1;
		for (int i=1; i<n; i++)
			s += 1 - use_machine(new int[] {0, i});
		return s;
	}
}

File: ioi2020-mushrooms/solution/trivial-one-by-one.py
#
# Trivial one-by-one solution
#
# Author: Kian Mirjalali
#
from stub import use_machine

def count_mushrooms(n: int) -> int:
    s = 1
    for i in range(1, n):
        s += 1 - use_machine([0, i])
    return s
